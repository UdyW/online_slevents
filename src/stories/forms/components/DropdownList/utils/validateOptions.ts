export const validateOptions = (options) =>
  Array.isArray(options) && (options !== undefined || options.length !== 0);
