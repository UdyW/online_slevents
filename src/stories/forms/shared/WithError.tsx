/* TODO fix the below */
/* eslint-disable @typescript-eslint/ban-types */
import React from 'react';
import styled from 'styled-components';
import { useThemeFallback } from '../../theming';
import { WithErrorProps } from '../types';

const StyledWithError = styled.div`
  &.has-error {
    padding-left: ${(p) => p.theme.spacing.xs};
    border-left: 2px solid ${(p) => p.theme.borderColors.borderColor2};
  }
`;

const ErrorMessage = styled.p`
  display: block;
  margin: ${(p) => p.theme.spacing.xs2} 0 0;
  font-size: ${(p) => p.theme.fontSizes.sm};
  color: ${(p) => p.theme.fontColors.fontColor1};
`;

const WithError =
  <P extends object>(
    Component: React.ComponentType<P>,
  ): React.FC<P & WithErrorProps> =>
  // eslint-disable-next-line react/display-name
  ({
    hasError = false,
    errorMessage = 'Required',
    theme: propsTheme,
    ...props
  }) => {
    // TODO fix the below
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const theme = useThemeFallback(propsTheme);
    const mergedProps = {
      ...props,
      hasError,
    };

    return (
      <StyledWithError className={hasError ? 'has-error' : ''} theme={theme}>
        <Component {...(mergedProps as P & WithErrorProps)} />

        {hasError && (
          <ErrorMessage
            id={`${props.id}-error`}
            className="error-message"
            data-qa="with-error__error-message"
            theme={theme}
          >
            {errorMessage}
          </ErrorMessage>
        )}
      </StyledWithError>
    );
  };

export { WithError };
