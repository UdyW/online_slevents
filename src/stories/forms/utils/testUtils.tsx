import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { createTheme } from '../../theming';
import 'jest-styled-components';

export const mountWithTheme = (children: React.ReactChild): ReactWrapper =>
  mount(<ThemeProvider theme={createTheme()}>{children}</ThemeProvider>);
