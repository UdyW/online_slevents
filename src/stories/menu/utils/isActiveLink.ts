import { GetIsActiveRoute } from '../types';

const isActiveLink = (
  getIsActiveRoute: GetIsActiveRoute,
  currentRoute = '',
  currentUrl = '',
  isPrimaryMenuItem?: boolean,
): boolean => {
  const url = isPrimaryMenuItem ? currentUrl.split('?')[0] : currentUrl;

  return getIsActiveRoute
    ? getIsActiveRoute(currentRoute, currentUrl)
    : url === currentRoute;
};

export { isActiveLink };
