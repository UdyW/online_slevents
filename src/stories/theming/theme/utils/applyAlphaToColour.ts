const clampAlphaValue = (value: number): number => {
  if (value < 0) {
    return 0;
  }

  if (value > 1) {
    return 1;
  }

  return value;
};

const hexToRgb = (colour: string): string => {
  const newcolour = colour?.substring(1) || '';

  if (newcolour) {
    const re = new RegExp(`.{1,${newcolour.length / 3}}`, 'g');

    let colours: Array<string> = newcolour.match(re);

    if (colours && colours[0].length === 1) {
      colours = colours.map((n) => n + n);
    }

    return colours
      ? `rgb(${colours.map((n) => parseInt(n, 16)).join(', ')})`
      : '';
  }

  return '';
};

/**
 * Apply alpha to an rbg or hex colour
 *
 * @example
 * ```
 * applyAlphaToColour('#000', 0.5);
 * applyAlphaToColour('#000000', 0.5);
 * applyAlphaToColour('rgb(1, 1, 1)', 0.5);
 * applyAlphaToColour(theme.palette.primary.primary1, 0.5);
 * ```
 *
 * @param {string} colour - an rgb or hex colour
 * @param {number} alpha - a value between 0 and 1
 */
const applyAlphaToColour = (colour: string, alphaValue: number): string => {
  if (colour && alphaValue) {
    let rgbColour = colour;

    if (rgbColour.charAt(0) === '#') {
      rgbColour = hexToRgb(rgbColour);
    }

    const marker = rgbColour.indexOf('(');

    const type = rgbColour.substring(0, marker);

    if (type !== 'rgb') {
      throw new Error(`applyAlphaToColour: unsupported colour ${colour}`);
    }

    const colourValues = rgbColour
      .substring(marker + 1, rgbColour.length - 1)
      .split(',');

    const clampedAlpha = clampAlphaValue(alphaValue);

    return `rgba(${colourValues.join(',')}, ${clampedAlpha})`;
  }

  throw new Error(
    'applyAlphaToColour: the colour or alpha value provided is not valid',
  );
};

export { clampAlphaValue, hexToRgb, applyAlphaToColour };
