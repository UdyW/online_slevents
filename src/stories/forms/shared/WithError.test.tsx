import React from 'react';
import { WithError } from './WithError';
import { mountWithTheme } from '../utils/testUtils';

const TestComponent = () => <>Test</>;
const TestComponentWithError = WithError(TestComponent);

const props = {
  id: 'text-id',
  content: 'test content',
};

describe('WithError()', () => {
  test('should render correctly', () => {
    const target = mountWithTheme(<TestComponentWithError {...props} />);

    expect(target).toMatchSnapshot();
  });
});
