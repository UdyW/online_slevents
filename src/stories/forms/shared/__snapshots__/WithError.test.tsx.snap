// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithError() should render correctly 1`] = `
.c0.has-error {
  padding-left: 0.459rem;
  border-left: 2px solid #DD1D21;
}

<Component
  theme={
    Object {
      "backgroundColors": Object {
        "backgroundColor1": "#FBCE07",
        "backgroundColor10": "#404040",
        "backgroundColor11": "#BD2C2F",
        "backgroundColor2": "#DD1D21",
        "backgroundColor3": "#0097BB",
        "backgroundColor4": "#003C88",
        "backgroundColor5": "#008443",
        "backgroundColor6": "#FFFFFF",
        "backgroundColor7": "#F9F9F9",
        "backgroundColor8": "#D9D9D9",
        "backgroundColor9": "#595959",
      },
      "borderColors": Object {
        "borderColor1": "#FBCE07",
        "borderColor10": "#F9F9F9",
        "borderColor11": "#7F7F7F",
        "borderColor2": "#DD1D21",
        "borderColor3": "#0097BB",
        "borderColor4": "#008443",
        "borderColor5": "#D9D9D9",
        "borderColor6": "#A6A6A6",
        "borderColor7": "#595959",
        "borderColor8": "#404040",
        "borderColor9": "#FFFFFF",
      },
      "breakpoints": Object {
        "lg": "896px",
        "md": "768px",
        "sm": "640px",
        "up": [Function],
        "xl": "1080px",
        "xl2": "1600px",
        "xs": "480px",
      },
      "fontColors": Object {
        "fontColor1": "#DD1D21",
        "fontColor2": "#0097BB",
        "fontColor3": "#008443",
        "fontColor4": "#404040",
        "fontColor5": "#FFFFFF",
        "fontColor6": "#7F7F7F",
      },
      "fontSizes": Object {
        "lg": "1.138rem",
        "md": "1rem",
        "sm": "0.878rem",
        "xl": "1.296rem",
        "xl2": "1.476rem",
        "xl3": "1.68rem",
        "xl4": "1.913rem",
        "xl5": "2.178rem",
        "xl6": "2.479rem",
        "xs": "0.772rem",
      },
      "linkColors": Object {
        "linkColor1": "#003C88",
        "linkColor2": "#404040",
        "linkColor3": "#0A274D",
        "linkColor4": "#800080",
      },
      "palette": Object {
        "helpers": Object {
          "applyAlphaToColour": [Function],
        },
        "misc": Object {
          "misc1": "#811114",
          "misc2": "#BD2C2F",
          "misc3": "#62BFD0",
          "misc4": "#800080",
          "misc5": "#0A274D",
        },
        "neutral": Object {
          "black": "#000000",
          "grey1": "#F9F9F9",
          "grey2": "#D9D9D9",
          "grey3": "#A6A6A6",
          "grey4": "#7F7F7F",
          "grey5": "#595959",
          "grey6": "#404040",
          "white": "#FFFFFF",
        },
        "primary": Object {
          "primary1": "#FBCE07",
          "primary2": "#DD1D21",
          "primary3": "#0097BB",
        },
        "secondary": Object {
          "secondary1": "#003C88",
          "secondary2": "#008443",
          "secondary3": "#89CFDC",
        },
      },
      "spacing": Object {
        "lg": "1.476rem",
        "md": "1rem",
        "sm": "0.678rem",
        "xl": "2.178rem",
        "xl2": "3.213rem",
        "xl3": "4.742rem",
        "xl4": "6.997rem",
        "xl5": "10.326rem",
        "xs": "0.459rem",
        "xs2": "0.311rem",
        "xs3": "0.211rem",
      },
    }
  }
>
  <Component
    content="test content"
    id="text-id"
  >
    <styled.div
      className=""
      theme={
        Object {
          "backgroundColors": Object {
            "backgroundColor1": "#FBCE07",
            "backgroundColor10": "#404040",
            "backgroundColor11": "#BD2C2F",
            "backgroundColor2": "#DD1D21",
            "backgroundColor3": "#0097BB",
            "backgroundColor4": "#003C88",
            "backgroundColor5": "#008443",
            "backgroundColor6": "#FFFFFF",
            "backgroundColor7": "#F9F9F9",
            "backgroundColor8": "#D9D9D9",
            "backgroundColor9": "#595959",
          },
          "borderColors": Object {
            "borderColor1": "#FBCE07",
            "borderColor10": "#F9F9F9",
            "borderColor11": "#7F7F7F",
            "borderColor2": "#DD1D21",
            "borderColor3": "#0097BB",
            "borderColor4": "#008443",
            "borderColor5": "#D9D9D9",
            "borderColor6": "#A6A6A6",
            "borderColor7": "#595959",
            "borderColor8": "#404040",
            "borderColor9": "#FFFFFF",
          },
          "breakpoints": Object {
            "lg": "896px",
            "md": "768px",
            "sm": "640px",
            "up": [Function],
            "xl": "1080px",
            "xl2": "1600px",
            "xs": "480px",
          },
          "fontColors": Object {
            "fontColor1": "#DD1D21",
            "fontColor2": "#0097BB",
            "fontColor3": "#008443",
            "fontColor4": "#404040",
            "fontColor5": "#FFFFFF",
            "fontColor6": "#7F7F7F",
          },
          "fontSizes": Object {
            "lg": "1.138rem",
            "md": "1rem",
            "sm": "0.878rem",
            "xl": "1.296rem",
            "xl2": "1.476rem",
            "xl3": "1.68rem",
            "xl4": "1.913rem",
            "xl5": "2.178rem",
            "xl6": "2.479rem",
            "xs": "0.772rem",
          },
          "linkColors": Object {
            "linkColor1": "#003C88",
            "linkColor2": "#404040",
            "linkColor3": "#0A274D",
            "linkColor4": "#800080",
          },
          "palette": Object {
            "helpers": Object {
              "applyAlphaToColour": [Function],
            },
            "misc": Object {
              "misc1": "#811114",
              "misc2": "#BD2C2F",
              "misc3": "#62BFD0",
              "misc4": "#800080",
              "misc5": "#0A274D",
            },
            "neutral": Object {
              "black": "#000000",
              "grey1": "#F9F9F9",
              "grey2": "#D9D9D9",
              "grey3": "#A6A6A6",
              "grey4": "#7F7F7F",
              "grey5": "#595959",
              "grey6": "#404040",
              "white": "#FFFFFF",
            },
            "primary": Object {
              "primary1": "#FBCE07",
              "primary2": "#DD1D21",
              "primary3": "#0097BB",
            },
            "secondary": Object {
              "secondary1": "#003C88",
              "secondary2": "#008443",
              "secondary3": "#89CFDC",
            },
          },
          "spacing": Object {
            "lg": "1.476rem",
            "md": "1rem",
            "sm": "0.678rem",
            "xl": "2.178rem",
            "xl2": "3.213rem",
            "xl3": "4.742rem",
            "xl4": "6.997rem",
            "xl5": "10.326rem",
            "xs": "0.459rem",
            "xs2": "0.311rem",
            "xs3": "0.211rem",
          },
        }
      }
    >
      <div
        className="c0"
      >
        <TestComponent
          content="test content"
          hasError={false}
          id="text-id"
        >
          Test
        </TestComponent>
      </div>
    </styled.div>
  </Component>
</Component>
`;
